geomstats/discretized_curves_space.py

```diff
@@ -0,0 +1,437 @@
+                base_curve[:, 1:, :] - base_curve[:, :-1, :])
+        base_curve_velocity_norm = self.pointwise_norm(base_curve_velocity,
+                                                       base_curve[:, :-1, :])
+
+        coef_1 = 1 / gs.sqrt(base_curve_velocity_norm)
+        coef_2 = - 1 / (2 * base_curve_velocity_norm ** (5/2)) * \
+            self.pointwise_inner_product(tangent_vec_derivative,
+                                         base_curve_velocity,
+                                         base_curve[:, :-1, :])
+        srv_initial_derivative = gs.transpose(coef_1) * \
+            gs.transpose(tangent_vec_derivative, (2, 1, 0)) + \
+            gs.transpose(coef_2) * gs.transpose(base_curve_velocity, (2, 1, 0))
+        srv_initial_derivative = gs.transpose(srv_initial_derivative,
+                                              (2, 1, 0))
+
+        end_curve_srv = self.l2_metric.exp(tangent_vec=srv_initial_derivative,
+                                           base_curve=base_curve_srv)
+        end_curve_starting_point = self.embedding_metric.exp(
+                tangent_vec=tangent_vec[:, 0, :],
+                base_point=base_curve[:, 0, :])
+        end_curve = self.square_root_velocity_inverse(
+                end_curve_srv, end_curve_starting_point)
+
+        return end_curve
+
+    def log(self, curve, base_curve):
+        """
+        Riemannian logarithm of a curve wrt a base curve.
+        """
+        if not isinstance(self.embedding_metric, EuclideanMetric):
+            raise AssertionError('The logarithm map is only implemented '
+                                 'for dicretized curves embedded in a '
+                                 'Euclidean space.')
+        curve = gs.to_ndarray(curve, to_ndim=3)
+        base_curve = gs.to_ndarray(base_curve, to_ndim=3)
+        n_curves = curve.shape[0]
+        n_sampling_points = curve.shape[1]
+        n_coords = curve.shape[2]
+
+        curve_srv = self.square_root_velocity(curve)
+        base_curve_srv = self.square_root_velocity(base_curve)
+
+        base_curve_velocity = n_sampling_points * (
+                base_curve[:, 1:, :] - base_curve[:, :-1, :])
+        base_curve_velocity_norm = self.pointwise_norm(base_curve_velocity,
+                                                       base_curve[:, :-1, :])
+
+        coef_1 = gs.sqrt(base_curve_velocity_norm)
+        coef_2 = 1 / base_curve_velocity_norm ** (3/2) * \
+            self.pointwise_inner_product(curve_srv - base_curve_srv,
+                                         base_curve_velocity,
+                                         base_curve[:, :-1, :])
+        log_derivative = gs.transpose(coef_1) * \
+            gs.transpose(curve_srv - base_curve_srv, (2, 1, 0)) + \
+            gs.transpose(coef_2) * gs.transpose(base_curve_velocity, (2, 1, 0))
+        log_derivative = gs.transpose(log_derivative, (2, 1, 0))
+        log_0 = self.embedding_metric.log(point=curve[:, 0, :],
```

Comment:
maybe `_initial_sampling_points` instead of `_0` would make the code more readable for later?

Reply:
How about ```_initial_starting_points``` to be coherent with the input variable name of ```square_root_velocity_inverse``` ?